charts/plugin-barman-cloud/templates/server-certificate.yaml

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright The CloudNativePG Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+{{- if .Values.certificate.createServerCertificate }}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: barman-cloud-server
+  namespace: {{ include "plugin-barman-cloud.namespace" . }}
+spec:
+  commonName: barman-cloud
+  dnsNames:
+  - barman-cloud
```

Comment:
We should use the service name from the values here, once we do that the service name can be modified through values because a new certificate will be requested.